### DISCUSSION TITLE
SEO: CSR vs SSR

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#666] [Discussion] [answered]
Moved from Discord:

> I wanted to ask for some advise. I'm building react apps as a learning experience ahead of future client work and I love the fast loads that CSR offers. However I'd need to switch to SSG/SSR to boost SEO rankings (from my understanding). Sadly then I'd lose the fast load that makes react amazing. Any advice on how to have the best of both worlds?

**Comment by brillout** [comment, marked as answer]
You can do both CSR *and* SSR. Search engines only care about the first page rendering.

**Reply by Oliver-Turp** [reply]
Can you elaborate on how? :)

**Reply by samuelstroschein** [reply]
Server render a page and hydrate it on the client :)
